src/cache.toit

```diff
@@ -37,12 +37,20 @@ class Cache:
   /**
   Creates a new cache.
 
-  If the \$XDG_CACHE_HOME environment variable is set, the cache is located
-    at \$XDG_CACHE_HOME/$app-name. Otherwise, the cache will is stored
-    in \$(HOME)/.cache/$app-name.
+  Determines the cache directory in the following order:
+  - If APP_CACHE_DIR (where "APP" is the uppercased version of $app-name) is set,
+    uses it as the path to the cache.
+  - If the \$XDG_CACHE_HOME environment variable is set, the cache is located
+    at \$XDG_CACHE_HOME/$app-name.
+  - Otherwise, the cache directory is set to \$(HOME)/.cache/$app-name.
   */
   constructor --app-name/string:
     app-name-upper := app-name.to-ascii-upper
+    env := os.env
+    if env.contains "$(app-name-upper)_CACHE_DIR":
```

Comment:
Use `get --if-absent` or a null check?

Reply:
This isn't a real `Map`, so no `--if-absent`, but null-check works.
done.